corehq/apps/app_manager/util.py

```diff
@@ -572,3 +575,27 @@ def get_form_data(domain, app):
         module_meta['forms'] = forms
         modules.append(module_meta)
     return modules, errors
+
+
+def get_and_assert_practice_user_in_domain(practice_user_id, domain):
+    # raises PracticeUserException if CommCareUser with practice_user_id is not a practice mode user
+    #   or if user doesn't belong to domain
+    try:
+        user = CommCareUser.get(practice_user_id)
+        if not user.domain == domain:
+            raise ResourceNotFound
+    except ResourceNotFound:
+        raise PracticeUserException(_(
+            "Practice User with id {id} not found, please make sure you have not deleted this user").format(
+            id=practice_user_id
+        ))
+    if not user.is_demo_user:
+        raise PracticeUserException(_(
+            "User {username} is not a practice user, "
+            "please turn on practice mode for this user".format(username=user.username)))
+    if user.is_deleted():
+        raise PracticeUserException(_(
+            "User {username} has been deleted, you can't use that user as "
+            "practice user".format(username=user.username)
+        ))
+    return user
```

Comment:
(I assume you meant deactivate)
Not sure if it should matter. Currently it doesn't matter. @sheelio what should happen if a user tries to make a new version of the app of which the practice user is deactivated?